File: Lexicographically Smallest Palindrome.cpp
class Solution {
public:
    string makeSmallestPalindrome(string s) {
        int j = s.size() - 1;
        int i = 0;

        while (i < j) {
            if (s.at(j) < s.at(i)) {
                s.at(i) = s.at(j);
            } else {
                s.at(j) = s.at(i);
            }
            i++;
            j--;
        }

        return s;
    }
};

File: Lexicographically Smallest Palindrome.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        s = list(s)
        i, j = 0, len(s)-1
        while i < j:
            if s[i] != s[j]:
                c = min(s[i], s[j])
                s[i], s[j] = c, c
            i += 1
            j -= 1
        return ''.join(s)
